sdk/typescript/src/module/entrypoint/register.ts

```diff
@@ -122,11 +143,17 @@ export class Register {
    * Create a function in the Dagger API.
    */
   addFunction(fct: Method | DaggerInterfaceFunction): Function_ {
-    return dag
+    let fn = dag
       .function_(fct.alias ?? fct.name, addTypeDef(fct.returnType!))
       .withDescription(fct.description)
       .withSourceMap(addSourceMap(fct))
       .with(this.addArg(fct.arguments))
+
+    if (fct.deprecated) {
```

Comment:
Should be `if (fct.deprecated !== undefinied)` so we can also register function with empty deprecation reason like `/* @deprecated */`
(See my comments on `ast.ts`)